### ISSUE TITLE
check header guard variable for more extensions then "h"

### ISSUE THREAD

**Issue by dirk-thomas** [google/styleguide#79]
Currently the header guard variables are only checked for files ending with `.h` (https://github.com/google/styleguide/blob/2bfe5f4aa813e9294b3c3719d9648da90d84ac20/cpplint/cpplint.py#L4434).

When additional extensions are passed via the command line argument (e.g. `--extensions=c,cc,cpp,cxx,h,hh,hpp,hxx`) (some of) those should be checked too. This could either be done by passing the header extensions separately or consider all files which extension starts with `h`?

I am happy to provide a PR for this if there is a consensus how it should be addressed. I would suggest changing the referenced line to:

```
if file_extension.startswith('h'):
```

**Comment by dirk-thomas** [comment]
A similar check is being used in five locations. They would all benefit from checking with `startswith()` instead.

**Comment by LukeCz** [comment]
Added configurable option for any extension type in pull request #183.

**Comment by dirk-thomas** [comment]
Thank you for creating the new pull request with the new header option. In order to cover the use cases described here and the other PR (#88) it would be good if your PR would also allow to specify an option to override the extension of source files.

**Comment by LukeCz** [comment]
I fully agree, .cc should not be hardcoded either. I will add this option once/if this PR is merged, it should be easy to follow the same pattern :)
I do not want to code it immediately in case there are concerns regarding my approach and I will have to change.

**Comment by tkruse** [comment]
@LukeCz: In case it helps, in our fork of cpplint similar work has been done to consistently handle extensions: https://github.com/theandrewdavis/cpplint

We would love to have our work included upstream, but so far nobody showed interest in merging open PRs.

**Comment by LukeCz** [comment]
@tkruse: Ah, it seems we had a similar idea:) I would love for either one to be accepted to the official version so I do not have to merge the changes every time I download new version.

**Comment by LukeCz** [comment]
@dirk-thomas I think .cc is already addressed in the code - _IsSourceExtension() function - and my recent commit addresses .h.
It seems to me this case can be closed successfully.

**Comment by dirk-thomas** [comment]
@LukeCz I commented on the corresponding PR because I don't think the current code covers "enough" extensions: see https://github.com/google/styleguide/pull/88#issuecomment-250628631

**Comment by vapier** [comment]
public cpplint development in this repository has been halted (see https://github.com/google/styleguide/pull/837). please migrate to https://github.com/cpplint/cpplint now.
